runtime-sdk-macros/src/lib.rs

```diff
@@ -26,6 +43,45 @@ pub fn error_derive(input: TokenStream) -> TokenStream {
     error_derive::derive_error(input).into()
 }
 
+/// Derives traits from a non-trait `impl` block (rather than from a `struct`).
+///
+/// Only the `MethodHandler` trait is supported. In other words, given an
+/// `impl MyModule` block, the macro derives `impl MethodHandler for MyModule`.
+/// See also the `#[handler]` attribute.
+#[proc_macro_attribute]
+pub fn sdk_derive(args: TokenStream, input: TokenStream) -> TokenStream {
+    let input = syn::parse_macro_input!(input as syn::ItemImpl);
+    if args.to_string() == "MethodHandler" {
+        method_handler_derive::derive_method_handler(input).into()
+    } else {
+        panic!("#[sdk_derive] only supports #[sdk_derive(MethodHandler)]");
+    }
+}
+
+/// A helper attribute for `#[sdk_derive(...)]`. It doesn't do anyting on its own;
+/// it only mark functions that represent a paratime method handler.
+/// The permitted forms are:
+///  - `#[handler(call = "my_module.MyCall")]`: Marks a function that handles
+///        the "my_module.MyCall" call and can be passed to
+///        oasis_runtime_sdk::module::dispatch_call.
+///  - `#[handler(prefetch = "my_module.MyCall")]`: Marks a function that handles
+///        the request to prefetch any data ahead of the "my_module.MyCall" call.
+///        Its signature should be `Fn(
```

Comment:
This signature is clumsy to explain; compare to other `#[handler(...)]` versions. I'm sure people will turn to existing examples anyway to understand the types/usage, but: It would be nicer if there was an `oasis_runtime_sdk::module:dispatch_prefetch()` helper function, just like for the the other methods. It would parse `args` (just like the other `dispatch_*()` helper fns do) and would possibly expect a `Vec<Prefix>` from the user-implemented callback, rather than passing the unusual `add_prefix` fn.

I assumed that performance was the reason `prefetch` is an exception and doesn't have that kind of an ergonomic wrapper: Not every prefetch handler needs to parse args, and exposing the global `BTreeSet` of prefixes-to-fetch avoids some copying. LMK if that's the case or if I should update prefetch to match the other handlers more closely in style.

Reply:
Yes, only due to performance reasons, since the existing benchmarks (although in an unrealistic scenario) showed that the performance benefit of prefetch was minimal and the additional parsing / copying could lead to poorer overall performance.

Feel free to open an issue for this, once we have some more realistic benchmarks we could revisit this.